tensorpipe/core/pipe.cc

```diff
@@ -1264,13 +1405,54 @@ void Pipe::Impl::onReadWhileServerWaitingForBrochure_(
           impl.onAcceptWhileServerWaitingForChannel_(
               channelName, std::move(transport), std::move(connection));
         }));
-    channelRegistrationIds_[channelName] = token;
+    channelRegistrationIds_["cpu_" + channelName] = token;
```

Comment:
Could we just use two separate fields for cpu and cuda? It is quite cleaner than tampering with strings, and also statically safer.